### PR TITLE
Fix HTTP 500 error when verifying a new account

### PR DESCRIPTION
**Pull request by scottbommarito**
https://github.com/NuGet/NuGetGallery/issues/5889

When trying to confirm a different account than your own, it now says:
```
You cannot confirm a different user's email address!
Please log in as '<AccountName>' and then click the link again.
```

### CODE REVIEW

File: src/NuGetGallery/Views/Shared/Confirm.cshtml [hunk 1]
```diff
@@ -34,36 +34,28 @@
                     @<text>
                         @if (Model.WrongUsername)
                         {
+                            <span>You cannot confirm a different user's email address!</span>
```
Comment (scottbommarito) [inline src/NuGetGallery/Views/Shared/Confirm.cshtml]: The alert messages used to be on 3 lines.
```
<alert icon>
You cannot confirm a different user's email address!
Please log in as '@Model.AccountName' and then click the link again.
```

This changes it to:
```
<alert icon> You cannot confirm a different user's email address!
Please log in as '@Model.AccountName' and then click the link again.
```
which is more consistent with the rest of the site.

File: src/NuGetGallery/ViewModels/ConfirmationViewModel.cs [hunk 2]
```diff
@@ -24,6 +24,22 @@ public class ConfirmationViewModel
...
 
+        /// <summary>
+        /// Creates a <see cref="ConfirmationViewModel"/> for an attempt to confirm a different account's email.
+        /// </summary>
+        /// <param name="username">The username of the account whose email was attempted to be confirmed.</param>
+        public ConfirmationViewModel(string username)
+        {
+            AccountName = username;
+            WrongUsername = true;
+            SuccessfulConfirmation = false;
```
Comment (chenriksson) [inline src/NuGetGallery/ViewModels/ConfirmationViewModel.cs]: It's not apparent from the constructors that one is for an error case and the other isn't.

I recommend continuing to set these from outside the constructor.

Comment (scottbommarito) [reply src/NuGetGallery/ViewModels/ConfirmationViewModel.cs]: I agree, but I think I'd rather have
```
public static ConfirmationViewModel CreateForWrongUsername(string username)
{
    var model = new ConfirmationViewModel(username);
    ...
}
```
than setting it from outside the constructor.

**Comment by scottbommarito** [comment]
@chenriksson the 500 is that the constructor for `ConfirmationViewModel` expects a non-null `User`, and in the `UsersController`, when a username that's not the current user's username is provided, the `User` passed to `ConfirmationViewModel` is null. This PR introduces an additional constructor for `ConfirmationViewModel` that does not expect a non-null `User`.

File: src/NuGetGallery/ViewModels/ConfirmationViewModel.cs (hunk 2 above)
Comment (scottbommarito) [reply src/NuGetGallery/ViewModels/ConfirmationViewModel.cs]: Changed my mind, using your approach.

**Pull request merged**